### PR TITLE
Challenger rp2040

### PR DESCRIPTION
**Pull request by kenbell**
This board is a feather-rp2040 derivative, but does not have the UART on conventional pins, so move the UART definitions to per-board.  I don't have all boards, but where the docs/schematic suggest only a single UART is exposed, i've removed the definitions related to the other UART (typically UART0 is exposed, whereas UART1 may not be).

There's also a change to git ignore a file created by smoketest

### CODE REVIEW

**Comment by kenbell** [comment]
I'm working on a separate change to add support for the on-board LoRa chip to the drivers package.

**Comment by sago35** [comment]
image

https://ilabs.se/product/challenger-rp2040-lora/

**Comment by sago35** [comment]
@kenbell 
Could you add VID, PID, etc.
USB for rp2040 will be available soon.

ref : #2936

https://github.com/earlephilhower/arduino-pico/blob/master/boards.txt

```
# -----------------------------------
# iLabs Challenger 2040 LoRa
# -----------------------------------
challenger_2040_lora.vid.0=0x2e8a
challenger_2040_lora.pid.0=0x1023
challenger_2040_lora.build.vid=0x2e8a
challenger_2040_lora.build.pid=0x1023
challenger_2040_lora.build.usb_manufacturer="iLabs"
challenger_2040_lora.build.usb_product="Challenger 2040 LoRa"
```

**Comment by kenbell** [comment]
@sago35 - thanks for taking a look. I have pushed a new change with the USB info.

File: src/machine/board_feather_rp2040.go [hunk 1]
```diff
@@ -60,3 +65,35 @@ const (
...
+
+// UART on the RP2040
+var (
+	UART0  = &_UART0
+	_UART0 = UART{
+		Buffer: NewRingBuffer(),
+		Bus:    rp.UART0,
+	}
+
+	UART1  = &_UART0
```
Comment (sago35) [inline src/machine/board_feather_rp2040.go]: `&_UART1` ?

File: src/machine/board_challenger_rp2040.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+// UART on the RP2040
+var (
+	UART0  = &_UART0
+	_UART0 = UART{
+		Buffer: NewRingBuffer(),
+		Bus:    rp.UART0,
+	}
+
+	UART1  = &_UART0
```
Comment (sago35) [inline src/machine/board_challenger_rp2040.go]: `&_UART1` ?

File: src/machine/board_pico.go [hunk 3]
```diff
@@ -64,3 +69,35 @@ const (
...
+
+// UART on the RP2040
+var (
+	UART0  = &_UART0
+	_UART0 = UART{
+		Buffer: NewRingBuffer(),
+		Bus:    rp.UART0,
+	}
+
+	UART1  = &_UART0
```
Comment (sago35) [inline src/machine/board_pico.go]: `&_UART1` ?

**Review by sago35 — CHANGES_REQUESTED** [review]
There is a typo.
I think everything else is fine.

**Comment by kenbell** [comment]
@sago35 thanks for spotting those typos.  I've updated the PR

**Review by sago35 — APPROVED** [review]
LGTM

**Comment by kenbell** [comment]
It took a couple go-arounds to merge the xaio-rp2040 changes, but looks good now.

@deadprogram I think this is ready to merge to dev

**Comment by deadprogram** [comment]
Please rebase one more time against `dev`, please.

**Comment by kenbell** [comment]
thanks @deadprogram - rebased

**Comment by deadprogram** [comment]
Thanks @kenbell now merging.

**Pull request merged**